### PR TITLE
aws/eks-helm: Add rbac module to metrics-server

### PR DESCRIPTION
**Pull request by spartan-haopham**
## Summary
* Init `eks rbac` module
* Add `var.set_container_port`, `var.set_metrics_enabled` and `var.set_list_config` for dynamically create each set when being enabled

## Types of Changes

- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 🕷 Bug fix (non-breaking change which fixes an issue)
- [ ] 👏 Performance optimization (non-breaking change which addresses a performance issue)
- [ ] 🛠 Refactor (non-breaking change which does not change existing behavior or add new functionality)
- [ ] 📗 Library update (non-breaking change that will update one or more libraries to newer versions)
- [x] 📝 Documentation (non-breaking change that doesn't change code behavior, can skip testing)
- [ ] ✅ Test (non-breaking change related to testing)
- [ ] 🔒 Security awareness (changes that effect permission scope, security scenarios)

## Related Issues
https://www.notion.so/c0x12c/Infra-Rollout-the-new-Terraform-modules-Helm-to-Regimen-16401fb05bf1803ca209e1a7c0a96520?pvs=4

### CODE REVIEW

File: aws/eks-helm/metrics-server/examples/main.tf [hunk 1]
```diff
@@ -5,3 +5,17 @@ module "example" {
...
+
+  helm_release_name  = "metrics-server"
+  namespace          = "kube-system"
+  helm_chart_version = "3.12.2"
+
+  set_rbac_create = {
+    name  = "rbac.create"
+    value = true
+  }
+}
```
Comment (spartan-tainguyen) [inline aws/eks-helm/metrics-server/examples/main.tf:21]: EOF

File: aws/eks-helm/metrics-server/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 # Changelog
 
 All notable changes to this project will be documented in this file.
+## [0.1.28]() (2024-12-27)
+### Features
+* Init `eks rbac` module
```
Comment (spartan-tainguyen) [inline aws/eks-helm/metrics-server/CHANGELOG.md:6]: For clarification, could you explain the necessity of a sub-module for RBAC? The existing chart configuration already defaults `rbac.create` to true.

File: aws/eks-helm/metrics-server/main.tf [hunk 3]
```diff
@@ -8,8 +8,36 @@ resource "helm_release" "metrics_server" {
   keyring    = ""
 
   # If true, allow unauthenticated access to /metrics.
-  set {
-    name  = "metrics.enabled"
-    value = false
+  dynamic "set" {
+    for_each = var.set_metrics_enabled == null ? [] : [var.set_metrics_enabled]
```
Comment (spartan-tainguyen) [inline aws/eks-helm/metrics-server/main.tf:12]: can we group the 3 sets with something like
```tf
for_each = [for v in [var.set_metrics_enabed, var.set_container_port, var.set_rbac_create]: v if v != null]
```

Comment (spartan-haopham) [reply aws/eks-helm/metrics-server/main.tf:12]: I got it, thank you.

**Pull request closed without merging**